pkg/storageos/deploy.go

```diff
@@ -302,6 +310,40 @@ func (s *Deployment) deleteObject(obj runtime.Object) error {
 	return nil
 }
 
+// getDeploymentByName returns a StorageOS Deployment resource.
+func (s Deployment) getDeploymentByName(name string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: s.stos.Spec.GetResourceNS(),
+			Labels: map[string]string{
+				"app": "storageos",
+			},
+		},
+	}
+}
+
+// getStatefulSetByName returns a StorageOS StatefulSet resource.
+func (s Deployment) getStatefulSetByName(name string) *appsv1.StatefulSet {
+	return &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "StatefulSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: s.stos.Spec.GetResourceNS(),
+			Labels: map[string]string{
+				"app": "storageos",
+			},
+		},
+	}
+}
```

Comment:
Moved and renamed these functions to be more generic than just CSI helper specific.